src/BinaryInstaller.php

```diff
@@ -240,4 +240,17 @@ protected function installBinary($binary, $version, $url, $sha, $hashalgo = 'sha
             }
         }
     }
+
+    /**
+     * Return if a file needs to be downloaded or not.
+     *
+     * @param string $cacheDestination The destination path to the downloaded file.
+     * @param $hashalgo The hash algorithm used to validate the file.
+     * @param $hash The hash used to validate the file.
+     *
+     * @return bool True if the file needs to be downloaded again, false otherwise.
+     */
+    private function needsDownload(string $cacheDestination, $hashalgo, $hash): bool {
+        return !$this->cache->isEnabled() || !file_exists($cacheDestination) || (file_exists($cacheDestination) && hash_file($hashalgo, $cacheDestination) !== $hash);
+    }
```

Comment:
I think this should have failed as the existing code follows symfony block style. I'll investigate separately.